### PR TITLE
Fix compile warnings

### PR DESCRIPTION
**Pull request by MasterDuke17**
You only see the warnings the first commit fixes if FSA_DEBUG is enabled, but the second commit fixes ones always seen.

### CODE REVIEW

File: src/core/fixedsizealloc.c [hunk 1]
```diff
@@ -294,7 +294,7 @@ void MVM_fixed_size_free(MVMThreadContext *tc, MVMFixedSizeAlloc *al, size_t byt
 #ifdef MVM_VALGRIND_SUPPORT
         if (RUNNING_ON_VALGRIND) {
             char command[128];
-            snprintf(&command, 128, "check_memory defined %p %d", dbg, bytes + 8);
+            snprintf(command, 128, "check_memory defined %p %lu", dbg, bytes + 8);
```
Comment (niner) [inline src/core/fixedsizealloc.c:297]: Shouldn't that be %zu since bytes is a size_t?

File: src/core/fixedsizealloc.c [hunk 2]
```diff
@@ -341,7 +341,7 @@ void MVM_fixed_size_free_at_safepoint(MVMThreadContext *tc, MVMFixedSizeAlloc *a
     if (dbg->alloc_size != bytes) {
 #ifdef MVM_VALGRIND_SUPPORT
         if (RUNNING_ON_VALGRIND) {
-            char command[128]; snprintf(&command, 128, "check_memory defined %p %lu", dbg, bytes + 8); VALGRIND_MONITOR_COMMAND(command);
+            char command[128]; snprintf(command, 128, "check_memory defined %p %lu", dbg, bytes + 8); VALGRIND_MONITOR_COMMAND(command);
```
Comment (niner) [inline src/core/fixedsizealloc.c:344]: same here

File: src/core/fixedsizealloc.c (hunk 1 above)
Comment (MasterDuke17) [reply src/core/fixedsizealloc.c:297]: Guess so, `%ld` is what gcc suggested, but I made it `%lu` to be consistent with the format below. I can change both.

Comment (MasterDuke17) [reply src/core/fixedsizealloc.c]: Did that, but also changed `MVMFixedSizeAllocDebug->alloc_size` to be `size_t`.

**Review by niner — APPROVED** [review]

**Pull request merged**